### ISSUE TITLE
Levenshtien Transformer Performance

### ISSUE THREAD

**Issue by spprabhu** [pytorch/fairseq#1523]
Hi I ran the Levenshtein Transformer Model on Both Original and Distilled Dataset.However I GET bleu of 23.87 on DIstilled after 230000k steps and training using fp--16 on single GPU without using --freq-update for GPU Simulation and 23.68 on original dataset after 260000k on 4 GPUs without fp--16.

What will I need to do to achieve the paper performance of 25.20 and 27.3 using Teacher Rescorer.

ALso when I dont give Batch size while evaluation The BLEU4 is 23.87 and 23.68.It takes 250 as Default batch size. However if I mention Batch size 1 the BLEU DROPS to 4.16 and speed to 7.14 sentences/Second.

WHY SO?
@kahne @MultiPath @myleott 

Hereby Giving the outputs

Result with Batch size 1
| Translated 3003 sentences (38803 tokens) in 389.2s (7.72 sentences/s, 99.69 tokens/s)                                                                             
| Generate test with beam=1: BLEU4 = 4.30, 51.4/16.2/8.4/4.6 (BP=0.320, ratio=0.467, syslen=30151, reflen=64504)

Result with Default Batch size of  250 

| Translated 3003 sentences (84361 tokens) in 9.3s (321.44 sentences/s, 9030.04 tokens/s)                                                                           
| Generate test with beam=4: BLEU4 = 23.87, 56.5/29.9/17.7/10.9 (BP=1.000, ratio=1.000, syslen=64536, reflen=64506)

**Comment by MultiPath** [comment]
@spprabhu  We recently found there is a new bug introduced by a recent refactor for batch size=1 decoding. Please update the code.

**Comment by myleott** [comment]
Ah, the "needs triage" label gets added automatically by the new Issue templates

**Comment by lematt1991** [comment]
Sounds like 5d7a81099462e9f19715ce5fa37c03816a750e12 should resolve the problem.  Closing...

**Comment by alphadl** [comment]
@spprabhu Have you reproduced the result reported in the paper?

**Comment by spprabhu** [comment]
Yes, But I didn't get the same results.
I could translate 3003 sentences in 9.3 sec with test beam size 4 and BLEU4 = 23.87
But I could only train it for 230000 steps and I used 1V100 using --fp16
The training lasted for about 15 hrs

**Comment by spprabhu** [comment]
Also if you want to translate just a single sentence at a time the BLEU decreases significantly.

**Comment by spprabhu** [comment]
Hi @alphadl did you get the exact results
